### PR TITLE
feat(user-notification): Old data cleanup cronjob

### PR DESCRIPTION
**Pull request by rafnarnason**
This pr adds a cronjob feature which in accordance with the decision to only keep notifications until six months old deletes all table rows older than 6 months - using kubernetes cronjob ... plus updated readme

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/services/user-notification/src/app/modules/notifications/notifications.service.ts [hunk 1]
```diff
@@ -285,4 +286,30 @@ export class NotificationsService {
       )
     }
   }
+
+  @Cron('0 * * * *') // run once every hour
```
Comment (kksteini) [inline apps/services/user-notification/src/app/modules/notifications/notifications.service.ts:290]: Is it necessary to do this every hour?
If the resolution is for six months is it not sufficient f.x. to do it once a day with f.x: `@Cron('0 0 * * *')`

**Review by kksteini — APPROVED** [review]

File: apps/services/user-notification/src/app/modules/notifications/notifications.service.ts (hunk 1 above)
Comment (rafnarnason) [reply apps/services/user-notification/src/app/modules/notifications/notifications.service.ts:290]: true - I just want to see this working on dev hourly to begin with, and then followup PR changing the cron a once a day

Comment (AndesKrrrrrrrrrrr) [reply apps/services/user-notification/src/app/modules/notifications/notifications.service.ts:290]: Hey, just noticed this while browsing.

How about using '@hourly' or '@weekly'?

Comment (rafnarnason) [reply apps/services/user-notification/src/app/modules/notifications/notifications.service.ts]: splendid idea

File: apps/services/user-notification/src/app/app.module.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { SequelizeModule } from '@nestjs/sequelize'
 import { SequelizeConfigService } from './sequelizeConfig.service'
 import { AuthConfig, AuthModule } from '@island.is/auth-nest-tools'
 import { environment } from '../environments/environment'
+import { ScheduleModule } from '@nestjs/schedule'
```
[the remark this answers is not in the document]
Comment (rafnarnason) [reply apps/services/user-notification/src/app/app.module.ts]: Kubernetes Cron implemented, please review

File: package.json [hunk 3]
```diff
@@ -326,6 +326,7 @@
     "@graphql-codegen/typescript-react-apollo": "3.3.3",
     "@graphql-codegen/typescript-resolvers": "2.7.3",
     "@jeremybarbet/contentful-extension-scripts": "0.22.7",
+    "@nestjs/schedule": "4.0.0",
```
Comment (baering) [inline package.json:329]: Remove this dependency?

**Review by baering — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (rafnarnason) [reply package.json:329]: of course - well spotted

**Review by baering — APPROVED** [review]
Nice LGTM 👍

**Pull request merged**